libcaf_test/caf/test/reporter.hpp

```diff
@@ -72,7 +72,19 @@ class CAF_TEST_EXPORT reporter {
     = 0;
 
   virtual void
-  info(std::string_view msg, const detail::source_location& location)
+  print_info(std::string_view msg, const detail::source_location& location)
+    = 0;
+
+  virtual void
+  print_debug(std::string_view msg, const detail::source_location& location)
+    = 0;
+
+  virtual void
+  print_warning(std::string_view msg, const detail::source_location& location)
+    = 0;
+
+  virtual void
+  print_trace(std::string_view msg, const detail::source_location& location)
```

Comment:
```suggestion
```

Let's skip `trace` for now. Tracing should affect the indentation level, so we'll probably need to split it into begin/end functions and/or work with guards. We have no immediate use case yet, so let's kick this down the road.